src/azul/service/manifest_service.py

```diff
@@ -1394,7 +1394,13 @@ def command_lines(cls,
         file_options = [
             '--fail-early',  # Exit curl with error on the first failure encountered
             '--continue-at -',  # Resume partially downloaded files
-            '--retry 2',  # Retry a file download up to X times on transient error
+            # We set the retry count high here since it is possible for file
+            # downloads from a curl manifest to trip our WAF rate rule, and once
+            # tripped, it can take up to 5 min to clear (assuming no other
+            # requests from the offending IP were made). With a Retry-After of
+            # 30 sec, a minimum of 10 retries would be needed to ensure the
+            # client has waited long enough for the WAF rate rule to clear.
+            '--retry 15',  # Retry a file download up to X times on transient error
```

Comment:
This is extremely brittle. One change to the retry-after header or the rate rule and this could break. 

Extract all the relevant parameters into `azul.config` attributes and use them in a formula here.